modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/stat/StatisticsConfigurationTest.java

```diff
@@ -395,6 +395,50 @@ public void dropSingleColumnStatisticWhileNodeDown() throws Exception {
             (stats) -> stats.forEach(s -> assertNull("Invalid stats: " + stats, s.columnStatistics("A"))));
     }
 
+    /**
+     * Checks orphan records cleanup on activation doesn't lead to grid hanging.
+     * - Start the grid, create table and collect statistics.
+     * - Ensure statistics for the table exists.
+     * - Disable StatisticsManagerConfiguration to prevent configuration changes in metastorage.
+     * - Drop table.
+     * - Re-activate the grid.
+     * - Ensures statistics for the table was dropped as well.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testOrphanDataCleanup() throws Exception {
```

Comment:
Not consider me as boring guy - but ticked about dead lock and we discuss test with dead lock and here i see orphan clearing and can`t link it with initial formulation, need some explanations here ..